### PR TITLE
Don't restart cockpit.socket with cockpit_started: no; check role options in integration test

### PR DESCRIPTION
**Pull request by martinpitt**
Yay, my first actual bug fix -- spotted while writing the integration test.

### CODE REVIEW

File: tests/tests_config.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          [Session]
+          IdleTimeout = 60
+
+        dest: /etc/cockpit/cockpit.conf
+      check_mode: yes
+      register: result
+      failed_when: result.changed
+
+    - import_role:
+        name: tests-cleanup
```
Comment (richm) [inline tests/tests_config.yml:59]: Please don't use a role for this - just implement the cleanup tasks using a file like `tests/tasks/cleanup.yml` then use `include_tasks: tasks/cleanup.yml` in your test playbooks - using a role for this unnecessarily complicates things

**Comment by martinpitt** [comment]
The [content comparison failed in one test](https://dl.fedoraproject.org/pub/alt/linuxsystemroles/logs/linux-system-roles-cockpit-pull-linux-system-roles_cockpit-36-ba0d199-centos-8-20211019-050808/artifacts/ansible.log), and the message is useless as it does not show the current file. What I *really* want to do is something like this:
```yaml
    - name: test - fetch generated configuration file
      slurp:
        src: /etc/cockpit/cockpit.conf
      register: conf

    - name: test - expected configuration file
      assert:
        that: |
          "{{conf['content'] | b64decode }}" == "#\n# Ansible managed\n#\n[WebService]\nLoginTitle = hello world\n\n[Session]\nIdleTimeout = 60\n\n"
```

But this does not work:
```
TASK [test - expected configuration file] ******************************************************************
fatal: [c]: FAILED! => {"msg": "The conditional check '\"#\n# Ansible managed\n#\n[WebService]\nLoginTitle = hello world\n\n[Session]\nIdleTimeout = 60\n\n\" == \"#\\n# Ansible managed\\n#\\n[WebService]\\nLoginTitle = hello world\\n\\n[Session]\\nIdleTimeout = 60\\n\\n\"\n' failed. The error was: Invalid conditional detected: unterminated string literal (detected at line 1) (<unknown>, line 1)"}
```

I played around with `'` vs. `"`, `| string`, `| quote`, and other syntax variations, but I just can't get this to behave. It feels like it ought to be possible to compare a string that contains `\n` in Ansible? Maybe you have some idea here, Google doesn't help.

For now I pushed an approach with calling diff against a previously written "expected" file.

**Comment by martinpitt** [comment]
The [rhel-x failure](https://dl.fedoraproject.org/pub/alt/linuxsystemroles/logs/linux-system-roles-cockpit-pull-linux-system-roles_cockpit-36-388835f-rhel-x-20211019-061849/artifacts/ansible.log) looks like a transient networking error:
```
fatal: [/cache/rhel-x.qcow2]: FAILED! => {"changed": false, "msg": "Failed to download packages: Cannot download Packages/cryptsetup-2.4.0-2.el9_b.x86_64.rpm: All mirrors were tried", "results": []}
```

Retrying (empty force-push)

File: tests/tests_config.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  tasks:
+    - meta: flush_handlers
+
+    - name: test - socket is enabled  # noqa 303 command-instead-of-module
+      command: "systemctl is-enabled {{ __cockpit_daemon }}"
+      changed_when: false
+
+    - name: test - socket is not active
+      shell: |
```
Comment (richm) [inline tests/tests_config.yml:24]: better to use service_facts for this

File: tests/tests_default.yml [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
-- hosts: all
+- name: Test role with default options
+  hosts: all
   roles:
     - linux-system-roles.cockpit
 
   tasks:
     - name: test - socket is active  # noqa 303 command-instead-of-module
-      command: systemctl is-active cockpit.socket
+      command: "systemctl is-active {{ __cockpit_daemon }}"
```
Comment (richm) [inline tests/tests_default.yml:9]: service_facts

File: tests/tests_default.yml [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
       changed_when: false
 
     - name: test - socket is enabled  # noqa 303 command-instead-of-module
-      command: systemctl is-enabled cockpit.socket
+      command: "systemctl is-enabled {{ __cockpit_daemon }}"
```
Comment (richm) [inline tests/tests_default.yml:13]: service_facts

File: tests/tests_config.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          WebService:
+            LoginTitle: "hello world"
+          Session:
+            IdleTimeout: 60
+
+  tasks:
+    - meta: flush_handlers
+
+    - name: test - socket is enabled  # noqa 303 command-instead-of-module
+      command: "systemctl is-enabled {{ __cockpit_daemon }}"
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply tests/tests_config.yml:20]: No, can't -- this really only includes `.service` units, but we are looking at `cockpit.socket`. However, thanks for pointing out! This is certainly useful for other things.

File: tests/tests_default.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (martinpitt) [reply tests/tests_default.yml]: done.

File: tests/tests_config.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    - meta: flush_handlers
+
+    - name: test - socket is enabled  # noqa 303 command-instead-of-module
+      command: systemctl is-enabled {{ __cockpit_daemon }}
+      changed_when: false
+
+    - name: test - socket is not active  # noqa 303 command-instead-of-module
+      command: systemctl is-active {{ __cockpit_daemon }}
+      register: result
+      failed_when: result.rc == 0
```
Comment (richm) [inline tests/tests_config.yml:26]: https://docs.ansible.com/ansible/2.9/user_guide/playbooks_tests.html#task-results
`failed_when: result is succeeded`

**Review by richm — APPROVED** [review]
lgtm
just waiting for a few more test results

**Pull request merged**